src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala

```diff
@@ -3,242 +3,221 @@ package edu.berkeley.ce.sparkrocks
-    *         joint set.
+    * @param numPartitions Number of partitions desired
+    * @param jointSetSpan Number of joint sets to search at a time for partition generation
+    * @param seedBlocks Blocks that will serve as partitions
+    * @param remainingJoints Seq of joints that were not used in seed block generation
+    * @return Returns a tuple containing a Seq of blocks and a Seq of joints. The Seq of blocks
+    *         are the the desired partitions. The Seq of joints are the remaining joints that were
+    *         not used to generate the partitions.
     */
   @tailrec
-  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block, numProcessors: Int,
-                      remainingJoints: Seq[Joint] = Seq.empty[Joint]): (Seq[Joint], Seq[Joint]) = {
-    if (jointSets.isEmpty) {
-      throw new IllegalStateException("ERROR: Unable to find satisfactory seed joints")
-    }
-    // Check that joint set is persistent
-    if (jointSets.head(0).shape.isEmpty) {
-      val seedJointOption = findSeedJoints(jointSets.head, inputVolume, numProcessors, inputVolume.volume)
-
-      // Check whether joint set yields satisfactory seed joints. If it does not, check next joint set.
-      if (seedJointOption.isEmpty) {
-          searchJointSets(jointSets.tail, inputVolume, numProcessors, remainingJoints ++ jointSets.head)
+  def generateSeedBlocks(jointSets: Seq[Seq[Joint]], inputVolume: Block, numPartitions: Int,
+                         jointSetSpan: Int = 1, seedBlocks: Seq[Block] = Seq.empty[Block],
+                         remainingJoints: Seq[Joint] = Seq.empty[Joint]): (Seq[Block], Seq[Joint]) = {
+    if (jointSetSpan > jointSets.length) {
+      if (seedBlocks.isEmpty) {
+        throw new IllegalStateException("ERROR: Unable to find satisfactory seed joints")
       } else {
-        val seedJoints = seedJointOption.get
-        val remainingFromJointSet = jointSets.head.diff(seedJoints)
-        val allOtherJoints = remainingJoints ++ remainingFromJointSet ++ jointSets.tail.flatten
-        (seedJoints, allOtherJoints)
+        val volumes = seedBlocks.map(_.volume)
+        val avgVolume = volumes.foldLeft(0.0)(_ + _) / volumes.length
+        println("\nLoad Balance Info:")
+        println(s"Number of partitions: ${volumes.length}")
+        println(s"Average Volume: $avgVolume")
+        println(s"Max volume: ${volumes.max}")
+        println(s"Min volume: ${volumes.min}")
+        (seedBlocks, remainingJoints)
       }
     } else {
-      searchJointSets(jointSets.tail, inputVolume, numProcessors, remainingJoints ++ jointSets.head)
+      // Find joints that cut input volume into approximately equal pieces
+      val (partitionBlocks, leftOverJoints) = findSeedBlocks(jointSets, inputVolume, numPartitions, jointSetSpan)
+      // Sufficient partitions found
+      if (partitionBlocks.length == numPartitions) {
```

Comment:
Yeah, it's a pretty strict check. My thinking was that this will make users aware of the number of partitions that could be found with the `numPartitions` they requested. At this point they can choose to either rerun using a different value for `numPartitions` or use the `f` flag.

Is this reasonable or how do you think we should proceed on this?